### DISCUSSION TITLE
filesystem routing / exclude dist folder

### DISCUSSION THREAD

**Discussion by aheissenberger** [vikejs/vike#833] [Discussion]
How can I **exclude additional folders from filesystem routing** or define only the **pages** folder to be relevant for routing?

Currently, when I have files in the dist folder after building the project filesystem routing will include the dist folder and throw e.g. duplicated error page template errors:
```
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: [vite-plugin-ssr@0.4.118][Wrong Usage] Only one _error.page.js is allowed, but found several: /dist/server/entries/renderer/_error /renderer/_error
    at getErrorPageId (/workspace/bookingengine/node_modules/.pnpm/vite-plugin-ssr@0.4.118_vite@4.3.3/node_modules/vite-plugin-ssr/dist/cjs/shared/error-page.js:16:29)
    at Proxy.renderPage (/workspace/bookingengine/node_modules/.pnpm/vite-plugin-ssr@0.4.118_vite@4.3.3/node_modules/vite-plugin-ssr/dist/cjs/node/runtime/renderPage.js:57:65)
    at async eval (/workspace/bookingengine/server/index.ts:37:25)
```

**Comment by brillout** [comment]
That shouldn't happen. What are the filenames of all your `dist/` files?

**Comment by aheissenberger** [comment]
Here is a basic ssr-vavite setup which shows the problem:
https://github.com/aheissenberger/vavite-node-loader-node20-fail

start with:
`pnpm build`

then run:
`pnpm dev` and open http://localhost:3000/

```

> @vavite/example-vite-plugin-ssr@ dev /Users/ah/SVN-Checkouts/TST/vite-plugin-ssr
> vavite dev

@vavite/reloader: Loading server entry

  vavite: 1.7.0 (vite: 4.3.3) development server is running (ready in 232 ms)

  ➜  Local:   http://localhost:3000/
  ➜  Network: use --host to expose
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: [vite-plugin-ssr@0.4.118][Wrong Usage] Only one _error.page.js is allowed, but found several: /dist/server/entries/dist/server/entries/renderer/_error /dist/server/entries/renderer/_error /renderer/_error
    at getErrorPageId (/Users/ah/SVN-Checkouts/TST/vite-plugin-ssr/node_modules/.pnpm/vite-plugin-ssr@0.4.118_vite@4.3.3/node_modules/vite-plugin-ssr/dist/cjs/shared/error-page.js:16:29)
    at Proxy.renderPage (/Users/ah/SVN-Checkouts/TST/vite-plugin-ssr/node_modules/.pnpm/vite-plugin-ssr@0.4.118_vite@4.3.3/node_modules/vite-plugin-ssr/dist/cjs/node/runtime/renderPage.js:57:65)
    at async render (/server/render.ts:9:23)

Node.js v20.0.0
 ELIFECYCLE  Command failed with exit code 1.
```

**I found the problem** not related to **vite-plugin-ssr** - it was the bundling with `vavite` which included the `dist` folder.
A simple exclude in the `tsconfig.json` fixed the problem - thanks for your help!
```json
{
	"compilerOptions": {
		...
	},
	"exclude": [
		"dist"
	]
}
```

**Reply by brillout** [reply]
Just hav a quick look at your reproduction. It's most likely due to your Rollup configuration.

Do you need to change `entryFileNames`?

**Comment by aheissenberger** [comment]
I have to reopen - my change in the `tsconfig.json` fixied the inclusion of `dist` into `dist` folder but does not fix the problem with the error of including the `dist` folder as valid filesystem routing folder 🤔

**Reply by brillout** [reply]
Are you (or one of your dependency) changing Rollup configuration? Normally, the generated filenames in `dist/` don't match VPS's globbing. Hence my question what are the filenames of your `dist/` files?

**Reply by brillout** [reply]
Btw. I agree that `dist/` should be ignored. It's a Vite issue and I recommend to make a Vite PR (should be easy to fix). CC me in the PR if make one.
